src/CommandGenerator/CommandGenerator.cs

```diff
@@ -280,7 +283,27 @@ private void AddParametersToCommand(DbCommand cmd, IEnumerable<ColumnInfo> colum
         /// <inheritdoc/>
         public object GetColumnValue(ColumnInfo columnInfo, T item)
         {
-            var value = columnInfo.PropertyInfo.GetValue(item, null);
+            var dynamicMethodArgs = new Type[] { typeof(T) };
+            var dynamicMethod = new DynamicMethod("GetColumnValue", typeof(object), dynamicMethodArgs);
+            ILGenerator ilGenerator = dynamicMethod.GetILGenerator();
+
+            ilGenerator.Emit(OpCodes.Ldarg_0);
+
+            MethodInfo fnGetValue = typeof(T).GetProperty(
+                columnInfo.PropertyInfo.Name, BindingFlags.Public | BindingFlags.Instance).GetGetMethod();
+
+            ilGenerator.Emit(OpCodes.Callvirt, fnGetValue);
+
+            if (columnInfo.PropertyInfo.PropertyType.IsValueType)
+            {
+                ilGenerator.Emit(OpCodes.Box, columnInfo.PropertyInfo.PropertyType);
+            }
+
+            ilGenerator.Emit(OpCodes.Ret);
+
+            var invoke = dynamicMethod.CreateDelegate(typeof(_getColumnValueDelegate)) as _getColumnValueDelegate;
+            var value = invoke(item);
```

Comment:
Delegate can by cached like in `DynamicMethodModelFactory`. Key will be generated probably from `T` and property name.